src/snowflake/connector/connection.py

```diff
@@ -439,12 +441,7 @@ def __init__(
 
         # get the imported modules from sys.modules
         self._log_telemetry_imported_packages()
-
-    def __del__(self) -> None:  # pragma: no cover
-        try:
-            self.close(retry=False)
-        except Exception:
-            pass
+        atexit.register(self._close_at_exit)
```

Comment:
Should there be a comment with the long term ticket number so it's clear this is a temporary solution?

Reply:
updated